miden-lib/src/transaction/inputs.rs

```diff
@@ -276,16 +287,22 @@ fn add_account_to_advice_inputs(
 /// - inputs_hash |-> inputs
 /// - asset_hash |-> assets
 /// - notes_hash |-> combined note data
-fn add_input_notes_to_advice_inputs(notes: &InputNotes, inputs: &mut AdviceInputs) {
+fn add_input_notes_to_advice_inputs(
+    notes: &InputNotes,
+    note_args: Option<BTreeMap<NoteId, Word>>,
```

Comment:
This could also probably be `tx_args: &TransactionInputs` and then we could use `tx_args.get_note_args(note.id()).unwrap_or_default()` internally.